mirdata/datasets/jtd.py

```
@@ -0,0 +1,722 @@
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR 
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+
+
+class Track(core.Track):
+    """JTD track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        audio_path (str): path to audio file
+        onsets_path (str): path to onsets file
+        midi_path (str): path to MIDI file
+        beats_path (str): path to beats file
+        instrument (str): name of the instrument for this track, either "piano", "bass", or "drums"
+
+    Properties:
+        audio(tuple): audio signal and sample rate for the isolated instrument track of this performance
+        musician (str): name of the musician playing the `instrument` on this track
+
+    Cached Properties:
+        beats (BeatData): beat times for this instrument
+        onsets (EventData): onset and offset times
+        midi (NoteData): midi pitches, onset, offset times, and velocities
+
+    """
+
+    def __init__(self, track_id, data_home, dataset_name, index, metadata):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name=dataset_name,
+            index=index,
+            metadata=lambda: json.load(open(self.get_path("metadata"), "r")),
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.onsets_path = self.get_path("onsets")
+        self.midi_path = self.get_path("midi")
+        self.beats_path = self.get_path("beats")
+        self.instrument = self._get_instrument()
+
+    def _get_instrument(self) -> Optional[str]:
+        """Helper function to get the name of the instrument for this track from the track ID"""
+        instrument = self.track_id.split("_")[1]
+        return instrument if instrument in ["piano", "bass", "drums"] else None
```

I see quite a few cases where you return `None` if there's no match. In some of these cases, it seems like a failure to find a match would indicate an error. Are you sure it's a good idea to silently suppress such errors?

---

See https://github.com/mir-dataset-loaders/mirdata/pull/652#discussion_r1900701582 --- given how the metadata and track IDs are constructed, I don't think there's any possibility that `instrument` will not be `in ["piano", "bass", "drums"]`, so this logic is redundant. So we can just do:

```
def _get_instrument(self) -> str:
    """Helper function to get the name of the instrument for this track from the track ID"""
    return self.track_id.split("_")[1]
```

---

I see that you are taking the instrument from the track id, so I guess there no chance that instrument is `None`, right? Then `return self.track_id.split("_")[1]` sounds good to me!

---

I think you could even have `self.instrument = track_id.split("_")[1]` in the `__init__`, to simplify.

---

> I think you could even have self.instrument = track_id.split("_")[1] in the __init__, to simplify.

Fixed in [0679cc8](https://github.com/mir-dataset-loaders/mirdata/pull/652/commits/0679cc8d045ab9e1666279087395bba48f93e6f8)